openhands/events/stream.py

```diff
@@ -62,6 +63,18 @@ class EventStream:
     _cur_id: int = 0
     _lock: threading.Lock = field(default_factory=threading.Lock)
 
+    def __init__(self, sid: str, file_store: FileStore, num_workers: int = 1):
+        self.sid = sid
+        self.file_store = file_store
+        self._queue: Queue[Event] = Queue()
+        self._thread_pools: dict[str, dict[str, ThreadPoolExecutor]] = {}
```

Comment:
I wonder if a shared open ended thread pool executor would make more sense than an executor per subscriber?

Reply:
If we are gonna differentiate based on subscriber, why use a full thread pool executor rather than simply a thread?

Reply:
I tried that, but it's too easy for one subscriber to block the others. My main goal here is to make it so subscriber A can't mess with subscriber B